### PR TITLE
[AssetMapper][Validator] Fix tests on Windows

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.0
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Part of https://github.com/symfony/symfony/issues/52206
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function testMaxSize($maxSize, $bytes, $binaryFormat)
     {
         $file = new File(['maxSize' => $maxSize]);
 
-        $this->assertSame($bytes, $file->maxSize);
+        $this->assertEquals($bytes, $file->maxSize);
```
Comment (alexandre-daubois) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: This test fails on Appveyor when dealing with the "4G" and "4g" datasets. I suspect PHP to cast 4 billion to a string when running on a 32bits architecture

Comment (stof) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: Can we make it 3G instead, to fit on 32bits ?

Comment (alexandre-daubois) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: That wouldn't fit, but good idea! But I set it to 2G to fit signed integers.

Another solution would be to use `assertEquals` and `assertSame` depending on the architecture but I don't really like it

File: src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php [hunk 2]
```diff
@@ -54,8 +55,9 @@ public function testLoadConfig()
     public function testSaveConfig()
     {
         $reader = new CompiledAssetMapperConfigReader($this->writableRoot);
-        $this->assertEquals($this->writableRoot.\DIRECTORY_SEPARATOR.'foo.json', realpath($reader->saveConfig('foo.json', ['foo' => 'bar'])));
-        $this->assertEquals(['foo' => 'bar'], json_decode(file_get_contents($this->writableRoot.'/foo.json'), true));
+
+        $this->assertSame(Path::normalize($this->writableRoot.\DIRECTORY_SEPARATOR.'foo.json'), Path::normalize(realpath($reader->saveConfig('foo.json', ['foo' => 'bar']))));
```
Comment (stof) [inline src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php:59]: Why normalizing paths here ? Comparing 2 real paths should not require any normalization.

Comment (alexandre-daubois) [reply src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php:59]: `CompiledAssetMapperConfigReader` does some internal operations on the directory string (in `saveConfig()`) which leads to a path with mixed `/` and `\`.

Comment (xabbuh) [reply src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php:59]: Is this still an issue after #52213?

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php [hunk 3]
```diff
@@ -101,8 +101,8 @@ public static function provideValidSizes()
             ['3M', 3000000, false],
             ['1gi', 1073741824, true],
             ['1GI', 1073741824, true],
-            ['4g', 4000000000, false],
-            ['4G', 4000000000, false],
+            ['2g', 2000000000, false],
+            ['2G', 2000000000, false],
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php:105]: Should we make this change on 5.4 to avoid conflicts when merging up?

Comment (alexandre-daubois) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php:105]: I agree, I'm opening the PR 👍

File: src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php (hunk 2 above)
Comment (stof) [reply src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php:59]: But the assertion is not on the return value of `saveConfig`. It is on the result of `realpath` for that return value.

Comment (alexandre-daubois) [reply src/Symfony/Component/AssetMapper/Tests/CompiledAssetMapperConfigReaderTest.php:59]: It doesn't @xabbuh, I had a look at an outdated Appveyor log. Closing this PR, thanks for the review!

**Comment by alexandre-daubois** [comment]
Closing, see https://github.com/symfony/symfony/pull/52258#discussion_r1369975865

**Pull request closed without merging**